app/app.go

```diff
@@ -1318,35 +1319,22 @@ func (d *Deployments) getNewDeploymentForDevice(ctx context.Context,
 		lastDeployment = deviceDeployment.Created
 	}
 
-	//get deployments newer then last device deployment
-	//iterate over deployments and check if the device is part of the deployment or not
-	for skip := 0; true; skip += 100 {
-		deployments, err := d.db.FindNewerActiveDeployments(ctx, lastDeployment, skip, 100)
-		if err != nil {
-			return nil, nil, errors.Wrap(err,
-				"Failed to search for newer active deployments")
-		}
-		if len(deployments) == 0 {
+	deployment, err := d.db.FindOldestActiveDeploymentForDevice(ctx, deviceID, lastDeployment)
+	if err != nil {
+		if err == mdriver.ErrNoDocuments {
 			return nil, nil, nil
 		}
+		return nil, nil, errors.Wrap(err,
+			"Failed to search for newer active deployments")
+	}
```

Comment:
Following from suggestion above.
```suggestion
	} else if deployment == nil {
		return nil, nil, nil
	}
```